app/scripts/modules/core/cluster/filter/clusterFilter.component.ts

```diff
@@ -0,0 +1,103 @@
+  CLUSTER_FILTER_SERVICE,
+  require('./clusterFilter.model.js'),
+  require('../../filterModel/dependentFilter/dependentFilter.service.js'),
+  require('./clusterDependentFilterHelper.service.js'),
+]);
+
+class ClusterFilterCtrl {
+  public categoryHeadings: any;
+  public stackHeadings: string[];
+  public instanceTypeHeadings: any;
+  public regionHeadings: any;
+  public accountHeadings: any;
+  public providerTypeHeadings: any;
+  public availabilityZoneHeadings: any;
+  public app: Application;
+  public tags: any[];
+
+  constructor(public $scope: IScope,
+              public clusterFilterService: any,
+              public ClusterFilterModel: any,
+              public $rootScope: IScope,
+              public clusterDependentFilterHelper: any,
+              public dependentFilterService: any,
+  ) {
+  }
+
+  public $onInit() {
+    const { $scope, $rootScope, ClusterFilterModel, clusterFilterService, app } = this;
+
+    $scope.sortFilter = ClusterFilterModel.sortFilter;
+    this.tags = ClusterFilterModel.tags;
+
+    if (app.serverGroups.loaded) {
+      this.initialize();
+    }
+
+    app.serverGroups.onRefresh($scope, () => this.initialize());
+
+    $scope.$on('$destroy', $rootScope.$on('$locationChangeSuccess', () => {
+      ClusterFilterModel.activate();
+      clusterFilterService.updateClusterGroups(app);
+    }));
+  }
+
+  public updateClusterGroups() {
+    const { dependentFilterService, ClusterFilterModel, clusterDependentFilterHelper, clusterFilterService, app } = this;
+
+    let {providerType, instanceType, account, availabilityZone, region} = dependentFilterService.digestDependentFilters({
+      sortFilter: ClusterFilterModel.sortFilter,
+      dependencyOrder: ['providerType', 'account', 'region', 'availabilityZone', 'instanceType'],
+      pool: clusterDependentFilterHelper.poolBuilder(app.serverGroups.data)
+    });
+
+    this.providerTypeHeadings = providerType;
+    this.accountHeadings = account;
+    this.availabilityZoneHeadings = availabilityZone;
+    this.regionHeadings = region;
+    this.instanceTypeHeadings = instanceType;
+
+    ClusterFilterModel.applyParamsToUrl();
+    clusterFilterService.updateClusterGroups(app);
+  };
+
+  private getHeadingsForOption(option: string) {
+    return compact(uniq(map(this.app.serverGroups.data, option) as string[])).sort();
+  }
+
+  public clearFilters() {
+    const {clusterFilterService, app} = this;
+    clusterFilterService.clearFilters();
+    clusterFilterService.updateClusterGroups(app);
+    this.updateClusterGroups();
+  }
+
+  public initialize() {
+    this.stackHeadings = ['(none)'].concat(this.getHeadingsForOption('stack'));
+    this.categoryHeadings = this.getHeadingsForOption('category');
+    this.$scope.clusters = this.app.clusters;
+    this.updateClusterGroups();
+  };
+}
+
+ngmodule.component('clusterFilter', {
+  templateUrl: require('./clusterFilter.component.html'),
+  controller: ClusterFilterCtrl,
+  controllerAs: 'clustersFilters',
```

Comment:
since this is an actual component, we should omit `controllerAs` and use `$ctrl` in the template